pkg/monitor/cluster/nodes.go

```diff
@@ -0,0 +1,57 @@
+package cluster
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type metricsLabel string
+
+const (
+	notReadyNodes           metricsLabel = "notReadyNodes"
+	memoryPressureNodes     metricsLabel = "memoryPressureNodes"
+	diskPressureNodes       metricsLabel = "diskPressureNodes"
+	pidPressureNodes        metricsLabel = "pidPressureNodes"
+	networkUnavailableNodes metricsLabel = "networkUnavailableNodes"
+)
+
+var metricsLabelMap = map[corev1.NodeConditionType]metricsLabel{
+	corev1.NodeReady:              notReadyNodes,
+	corev1.NodeMemoryPressure:     memoryPressureNodes,
+	corev1.NodeDiskPressure:       diskPressureNodes,
+	corev1.NodePIDPressure:        pidPressureNodes,
+	corev1.NodeNetworkUnavailable: networkUnavailableNodes,
+}
```

Comment:
I think maintaining this is avoidable?

Reply:
I can use `NodeConditionType` (https://godoc.org/k8s.io/api/core/v1#NodeConditionType) and concatenate it with some metric label string, this will help to get rid from the first half of this code (const above)
However it is convenient to have a structure that I can iterate over to cover all conditions, as well as it helps to invert the logic for the Node Ready condition and treat all in the same way

Reply:
:+1: to using the upstream type.